0004-dockerised-local-network.md

```diff
@@ -0,0 +1,68 @@
+```
+
+The Docker image can also be used in CI jobs to run end-to-end tests, particularly for tools that don't require fully fledged networks but still need to test integration with "close to real" networks. This significantly improves the developer experience (DX) and user experience (UX). Example candidates for this kind of testing include tools like [o1Sdk](https://github.com/o1-labs/snarkyjs) (formerly known as `SnarkyJS`), [zkapp-cli](https://github.com/o1-labs/zkapp-cli), and [Archive-Node-API](https://github.com/o1-labs/Archive-Node-API).
+
+The lightweight Mina Local Network created by the Docker image will have the following properties:
+
+- Transaction finality (`k`) is `30` blocks.
+- There are `720 slots` per `epoch`.
+- New blocks are produced every `~20-40` seconds.
+- Each block contains `5-8` transactions.
+- The network consumes `~4.5GB` of RAM at the beginning on `Linux` platform and `~5.5Gb` of RAM on `macOS`.
+- The startup and sync time is approximately `4` minutes.
+
+For more details on how to manually set up and use the lightweight Mina Network, please refer to the [Mina Local Network Manager](https://github.com/MinaProtocol/mina/tree/rampup/scripts/mina-local-network#mina-lightweight-network) documentation on GitHub.
+
+## Test plan and functional requirements
+
+1. **Testing goals and objectives**: The primary goal is to ensure that the Docker image can successfully spin up a Mina Local Network and that this network behaves as expected. This includes fast startup and syncing times, low resource consumption, and the ability to form a network with diverse types of participants.
+2. **Testing approach**: The testing will involve the manual functional testing only, at least for now, because of the low scripts complexity. Performance monitoring will also be conducted to ensure that the Docker image meets the resource efficiency requirements.
+3. **Testing scope**: The testing will cover all major functionalities of the Docker image, including Dockerization phase, and the resulting Mina Local Network operation.
+4. **Testing requirements**: The testing will require a suitable environment for running Docker, as well as tools for monitoring resource usage and blockchain behavior.
+5. **Testing resources**: The testing will be conducted on a local machine. The testing may also require additional tools for monitoring and debugging.
+
+## Drawbacks
+
+One drawback of this approach is the potential for discrepancies between the behavior of the lightweight Mina Local Network and the full Mina network. While the lightweight network is designed to be "close to real", it may not perfectly replicate all aspects of the full network, which could lead to misleading test results. To mitigate this drawback, we can state clearly in the documentation that the lightweight network is not intended to be a perfect replica of the full network and that users/developers should always double check their work against real networks.
```

Comment:
```suggestion
One drawback of this approach is the potential for discrepancies between the behavior of the lightweight Mina Local Network and the full Mina network. While the lightweight network is designed to be "close to real", it may not perfectly replicate all aspects of the full network, which could lead to misleading test results. To mitigate this drawback, we can state clearly in the documentation that the lightweight network is for local development only and is not intended to be a perfect replica of the full network. Mina Local Networks do not guarantee production-ready results. Users and developers must always double-check their work against real networks.
```

Reply:
Thank you, and I'll change the case to lower.